cli/internal/backends/nodejs/nodejs.go

```diff
@@ -34,6 +35,51 @@ var NodejsYarnBackend = api.LanguageBackend{
 	GetRunCommand: func() []string {
 		return []string{"yarn", "run"}
 	},
+	Detect: func(cwd string, backend *api.LanguageBackend) (bool, error) {
+		specfileExists := fs.FileExists(filepath.Join(cwd, backend.Specfile))
+		lockfileExists := fs.FileExists(filepath.Join(cwd, backend.Lockfile))
+		isNotBerry := !fs.PathExists(filepath.Join(cwd, ".yarn/releases"))
+
+		if specfileExists && lockfileExists && isNotBerry {
+			return true, nil
+		}
+
+		return false, nil
+	},
+}
+
+var NodejsBerryBackend = api.LanguageBackend{
+	Name:             "nodejs-berry",
+	Specfile:         "package.json",
+	Lockfile:         "yarn.lock",
+	FilenamePatterns: nodejsPatterns,
+	GetWorkspaceGlobs: func() ([]string, error) {
+		pkg, err := fs.ReadPackageJSON("package.json")
+		if err != nil {
+			return nil, fmt.Errorf("package.json: %w", err)
+		}
+		if len(pkg.Workspaces) == 0 {
+			return nil, fmt.Errorf("package.json: no workspaces found. Turborepo requires Yarn workspaces to be defined in the root package.json")
+		}
+		return pkg.Workspaces, nil
+	},
+	GetPackageDir: func() string {
+		return "node_modules"
+	},
+	GetRunCommand: func() []string {
+		return []string{"yarn", "run"}
+	},
+	Detect: func(cwd string, backend *api.LanguageBackend) (bool, error) {
+		specfileExists := fs.FileExists(filepath.Join(cwd, backend.Specfile))
+		lockfileExists := fs.FileExists(filepath.Join(cwd, backend.Lockfile))
+		isBerry := fs.PathExists(filepath.Join(cwd, ".yarn/releases"))
```

Comment:
This would match for **every** yarn project that enforces any version.

I made an small test in a project with yarn v1. If you enforce his version with `yarn policies set-version`, it will also have an file at `.yarn/releases`.

You could parse the version filename, like `yarn-1.22.17.cjs` -> `1.22.17` or run `yarn version` and get the output.

Reply:
Never knew, will fix this one!